### PR TITLE
Add sourcemap support

### PR DESCRIPTION
**Pull request by davidknezic**
This pull request adds basic sourcemap support as requested in #6:
- Inline and external sourcemaps
- Tests
- Docs

### CODE REVIEW

**Comment by davidknezic** [comment]
@sventschui @elessar-ch merge?

File: index.js [hunk 1]
```diff
@@ -3,20 +3,48 @@ var minimatch = require('minimatch');
...
-    processor.use(plugin);
-  });
+  var processor = postcss(plugins);
 
   return function (files, metalsmith, done) {
     var styles = Object.keys(files).filter(minimatch.filter("*.css", { matchBase: true }));
-    styles.forEach(function (file, index, arr) {
-      files[file].contents = new Buffer(processor.process(files[file].contents.toString()).css);
+
+    styles.forEach(function (file) {
```
Comment (sventschui) [inline index.js:24]: Not part of your change but we need to introduce an Promise.all() for all files before calling done();, or am I completely wrong?

Comment (davidknezic) [inline index.js:24]: You are right, it's very probable that processing is continued while the plugin already signalled that it's done.

**Comment by davidknezic** [comment]
Nice @sventschui, I think it's ready to merge now.

**Comment by davidknezic** [comment]
We'll just need to bump the version number, publish the package and close #6.

**Pull request merged**